### DISCUSSION TITLE
How to RAISE NOTICE with variable

### DISCUSSION THREAD

**Discussion by magick93** [supabase/supabase#9044] [Questions]
I'm trying to log to the `postgres_log` file to help debug pg functions.

I have the following `plpgsql` in a function:

```
SELECT is_production() INTO is_production;
RAISE NOTICE 'Is Production =  %', is_production;
```

I would expect to see something like `Is Production = true` in the log file. Instead I get `RAISE NOTICE 'Is Production =  %', is_production;`.

I've tried various other attempts to print out the value of a variable. And in every case - for example `RAISE NOTICE 'Calling send_email_nodemailer(%)', message;` - the entire line is printed to the file.

**Comment by GaryAustin1** [comment]
Try RAISE LOG with the same string and values.

**Reply by magick93** [reply]
Thanks, but unfortunately it produces the same result - variable name, not value.

**Reply by GaryAustin1** [reply]
With this:
```
begin
raise log 'TEST NOTICE %', 'kitty';
end;
```
I get this in my log
![Capturekitty](https://user-images.githubusercontent.com/54564956/191167860-849f4656-83ff-40ce-a3d7-887afd6264e0.JPG)

So not sure what is going on....

**Reply by magick93** [reply]
What are you using to view this?

I'm using, for example `grep 'nodemailer message' /var/log/postgresql/postgresql.csv`

**Reply by GaryAustin1** [reply]
I'm just using the Supabase hosted log in the dashboard database tab postgres log
![image](https://user-images.githubusercontent.com/54564956/191168613-8e423459-b13a-461e-ae86-4d8b4ae4db16.png)

**Reply by magick93** [reply]
Right.

I'm trying to debug my functions locally before pushing them the hosted version. So I'm using `grep` of the postgres log file in the postgres container.
